tooling/ast_fuzzer/src/program/func.rs

```diff
@@ -726,11 +726,14 @@ impl<'a> FunctionContext<'a> {
         max_depth: usize,
     ) -> arbitrary::Result<Option<TrackedExpression>> {
         // Collect the operations can return the expected type.
+        // Avoid operations that can fail in no-dynamic mode, otherwise they will be considered non-constant indexes.
         let ops = BinaryOp::iter()
             .filter(|op| {
                 types::can_binary_op_return(op, typ)
-                    && (!self.ctx.config.avoid_overflow || !types::can_binary_op_overflow(op))
-                    && (!self.ctx.config.avoid_err_by_zero || !types::can_binary_op_err_by_zero(op))
+                    && (!self.ctx.config.avoid_overflow && !self.in_no_dynamic
+                        || !types::can_binary_op_overflow(op))
+                    && (!self.ctx.config.avoid_err_by_zero && !self.in_no_dynamic
+                        || !types::can_binary_op_err_by_zero(op))
```

Comment:
Can parentheses be added to make it explicit what the `||` applies to? (though I don't know if clippy will complain about this, but we can silence clippy in that case)

Reply:
The idea was that this way it's `(!<should-avoid> || !<would-cause-problem>)`.

Changed to `!(<should-avoid-foo> || <should-avoid-bar>) || !<would-cause-problem>)`